### DISCUSSION TITLE
CheckoutLine and OrderLine Metadata

### DISCUSSION THREAD

**Discussion by ddedic** [saleor/saleor#8473] [General] [answered]
Is there any specific reason why `CheckoutLine` and `OrderLine` models do not extend `ModelWithMetadata`?

It would be really useful if we have meta support, for building custom functionality.
In my case I need to attach some user defined values of the Variant to the CartLine (and later to transfer it to OrderLine).

Any chance this could be implemented in the core?

**Comment by maarcingebala** [comment, marked as answer]
No specific reason, it just wasn't added to these models yet. But I think one of the community members will be working on that so I think we can expect it soon on the `main` branch.

**Reply by ddedic** [reply]
Great news. Thanks a lot!
